framework/doc/content/source/meshgenerators/BlockToMeshConverterGenerator.md

```diff
@@ -0,0 +1,29 @@
+# BlockToMeshConverterGenerator
+
+!syntax description /Mesh/BlockToMeshConverterGenerator
+
+## Overview
+
+The `BlockToMeshConverterGenerator` is for moving one (or more) blocks from a mesh to form a new mesh object.
+This does not erase/delete the blocks from the original mesh. The new mesh has only one block (named 0). 
```

Comment:
We're moving the input mesh to `mesh` below, right?  And then it gets deleted, as per usual `MeshGenerator` behavior?  But "This does not erase/delete the blocks from the original mesh." makes me imagine that the original mesh will instead somehow be preserved and available for use in a subsequent `MeshGenerator`.

I'd say delete that whole sentence, but maybe I'm just not yet understanding what was meant by it.

Reply:
It should be available for use in another MeshGenerator. That s how we did the sideset extrusion

Reply:
like the idea is that the mesh was not explicitly marked for deletion, and another MG will pick it up. Then whatever we did with the extracted blocks (extrusion or triangulation are planned rn) can be stitched on the initial mesh

Reply:
> It should be available for use in another MeshGenerator.

Sorry, but: can you explain how?

At `std::unique_ptr<MeshBase> mesh = std::move(_input);` the pointer to the input mesh is moved into `mesh`, which is a `unique_ptr` that still has an active deleter.  When `BlockToMeshConverterGenerator::generate()` ends, `mesh` goes out of scope, its destructor calls the deleter, and the input mesh gets deleted.  Is there something else I'm missing?

Reply:
seems like we should avoid the move then.

Reply:
Okay, I think I found the something else I'm missing.  In `MooseApp::executeMeshGenerators()`, we're checking how many downstream generators use each output mesh, and if we see N>1 of them then we make N-1 copies so that each can have its own, and then each downstream generator is responsible for deleting its own copy.  I can see ways to make that more efficient but I no longer see any problem with correctness.  Sorry about the false alarm!